SPHINXsys/src/shared/particle_dynamics/dissipation_dynamics/particle_dynamics_dissipation.hpp

```diff
@@ -427,6 +427,55 @@ namespace SPH
 		}
 	}
 	//=================================================================================================//
+	template <typename VariableType>
+	DampingPairwiseFromShell<VariableType>::
+		DampingPairwiseFromShell(BaseContactRelation &contact_relation, const std::string &variable_name, Real eta)
+		: LocalDynamics(contact_relation.sph_body_),
+		  DataDelegateContact<BaseParticles, ShellParticles>(contact_relation),
+		  eta_(eta), Vol_(particles_->Vol_), mass_(particles_->mass_),
+		  variable_(*particles_->getVariableByName<VariableType>(variable_name))
+	{
+		for (size_t k = 0; k != contact_particles_.size(); ++k)
+		{
+			shell_variable_.push_back(contact_particles_[k]->template getVariableByName<VariableType>(variable_name));
+		}
+	}
+	//=================================================================================================//
+	template <typename VariableType>
+	void DampingPairwiseFromShell<VariableType>::interaction(size_t index_i, Real dt)
+	{
+		Real Vol_i = Vol_[index_i];
+		Real mass_i = mass_[index_i];
+		VariableType &variable_i = variable_[index_i];
+
+		std::array<Real, MaximumNeighborhoodSize> parameter_b;
+
+		/** Contact interaction. */
+		for (size_t k = 0; k < contact_configuration_.size(); ++k)
+		{
+			StdLargeVec<VariableType> &variable_k = *(shell_variable_[k]);
+			Neighborhood &contact_neighborhood = (*contact_configuration_[k])[index_i];
+			// forward sweep
+			for (size_t n = 0; n != contact_neighborhood.current_size_; ++n)
+			{
+				size_t index_j = contact_neighborhood.j_[n];
+
+				parameter_b[n] = contact_particles_[k]->DegeneratedSpacing(index_j) * eta_ * contact_neighborhood.dW_ijV_j_[n] * Vol_i * dt / contact_neighborhood.r_ij_[n];
```

Comment:
what is "DegeneratedSpacing"? Could we have a more understandable name?

Reply:
Shell formulation is called Degenerated Approach, therefore, DegeneratedSpacing used here denoting the spacing being degenerated.

Reply:
What I thought is " shell particle is degenerated (having low standard or behavior) ) from volume particle in the thickness spacing. Then, with shell particle is interacting  with volume particle, the degenerated spacing is considered. 
If you have other suggestion, please let me know.

Reply:
Do you mean that you consider the shell particle as a squashed solid particle in the thickness dimension? If yes, I would just call it ThicknessAdjustedSpacing for example, much clearer I think.

Reply:
Name it "Thinkness" is one of my old choices, while this name is meaningless for volume particles as they do not have thickness. Maybe "ReducedThickness", "ReducedCrossSection", "DegeneratedThickness" or "DegeneratedCrossSection" is more readable.

Reply:
"ReducedThickness", "ReducedCrossSection" sounds good to me

Reply:
> "ReducedThickness", "ReducedCrossSection" sound good to me

Reply:
Ok my problem was more with spacing than degenerate, I went to look in the code and it is basically the reduced or degenerated _dimension_ (return 1 for volume, thickness for shells, and area for beams). 
I would either go by `crossSectionDimension` or `reducedDimension` (or their getXXX variant)

Also why is the naming convention changing ? It was using camelCase before, now part of the new code is in PascalCase. Please change uniformize back to camelCase

It is implemented by virtual function, which I am not so comfortable with, but I don't have a straightforward drop-in replacement to offer at the moment.